docs/forms.md

```diff
@@ -0,0 +1,120 @@
+---
+sidebar_label: "Forms"
+sidebar_position: 4
+---
+
+# Forms
+
+After finishing setting up flutter and stacked (if you didn't setup stacked check out [get-started](get-started.md)), one of the features stacked provides is form handling by connecting form data in view and view model so that business logic and UI are separate and consistent with our goal. 
+
+## Form Setup
+
+Before generating methods for handling the controllers and validation messages. You need to tell the generator what your form has. And to tell the generator we use [annotations](https://medium.com/swlh/dart-annotations-a-simple-intro-to-reflection-c654275cc967). 
+
+Go into the view file and At the top of your class add
+
+```dart
+@FormView(fields: [])
+```
+
+`fields` accepts Lists of `FormField` which can be:
+
+- **FormTextField** 
+- **FormDateField** 
+- **FormDropdownField** 
+
+Each of the items above accepts name which is required to generate an appropriate controllers and keys 
+
+As an example let's assume you have a folder named `sign_in` which has `sign_in_view.dart` and `sign_in_viewmodel.dart` on your sign_in_view it has two textfields username and password your annotaion will look like
+
+```dart
+@FormView(fields: [
+    FormTextField(name: 'username'),
+    FormTextField(name: 'password'),
+])
+```
+
+## Form Generating
+
+After adding the annotation go on your terminal and run 
+
+```shell
+flutter pub run build_runner build --delete-conflicting-outputs
+```
+
+
+The command will generate a `sign_in_view.form.dart` which has a mixin `$SignInView` holding TextEditingControllers, FocusNodes and extensions to set validation messages 
+
+in the previous example it will generate
```

Comment:
Combine this with the sentence above. Add a fullstop after messages and change this in to start with an uppercase letter I